runtime/lua/vim/snippet.lua

```diff
@@ -0,0 +1,535 @@
+local G = require('vim.lsp._snippet_grammar')
+local snippet_group = vim.api.nvim_create_augroup('vim/snippet', {})
+local snippet_ns = vim.api.nvim_create_namespace('vim/snippet')
+
+--- Returns the 0-based cursor position.
+---
+--- @return integer, integer
+local function cursor_pos()
+  local cursor = vim.api.nvim_win_get_cursor(0)
```

Comment:
It's a standard convention to cache these indexes as:

```lua
local api, fn = vim.api, vim.fn
```

And use these throughout.

Reply:
It's also standard not to do that; it's up to you.

Reply:
It usually depends how many references there are. If there's not many, then it's obviously not worth doing.

Reply:
As discussed in the chat: We don't seem to have an established convention yet so I'll leave it as is.